functions/zoom-downloader.py

```diff
@@ -1,240 +1,215 @@
 from datetime import datetime
-from urllib.parse import quote
 from collections import OrderedDict
-
 import logging
+
 logger = logging.getLogger()
 
-ZOOM_VIDEOS_BUCKET = env('ZOOM_VIDEOS_BUCKET')
-DOWNLOAD_QUEUE_NAME = env('DOWNLOAD_QUEUE_NAME')
-UPLOAD_QUEUE_NAME = env('UPLOAD_QUEUE_NAME')
-DEADLETTER_QUEUE_NAME = env('DEADLETTER_QUEUE_NAME')
+ZOOM_VIDEOS_BUCKET = env("ZOOM_VIDEOS_BUCKET")
+DOWNLOAD_QUEUE_NAME = env("DOWNLOAD_QUEUE_NAME")
+UPLOAD_QUEUE_NAME = env("UPLOAD_QUEUE_NAME")
+DEADLETTER_QUEUE_NAME = env("DEADLETTER_QUEUE_NAME")
 MIN_CHUNK_SIZE = 5242880
 MEETING_LOOKUP_RETRIES = 2
 MEETING_LOOKUP_RETRY_DELAY = 60
-ZOOM_API_BASE_URL = env('ZOOM_API_BASE_URL')
-ZOOM_ADMIN_EMAIL = env('ZOOM_ADMIN_EMAIL')
+ZOOM_ADMIN_EMAIL = env("ZOOM_ADMIN_EMAIL")
 DEFAULT_SERIES_ID = env("DEFAULT_SERIES_ID")
 CLASS_SCHEDULE_TABLE = env("CLASS_SCHEDULE_TABLE")
 LOCAL_TIME_ZONE = env("LOCAL_TIME_ZONE")
-
+DOWNLOAD_MESSAGES_PER_INVOCATION = env('DOWNLOAD_MESSAGES_PER_INVOCATION')
 # Recordings that happen within BUFFER_MINUTES a courses schedule
 # start time will be captured
 BUFFER_MINUTES = 30
 # Ignore recordings that are less than MIN_DURATION (in minutes)
 MINIMUM_DURATION = 2
 
-TIMESTAMP_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
-
 
 class PermanentDownloadError(Exception):
     pass
 
 
-class ApiLookupFailure(Exception):
-    """
-    Only raise this on retry-able failures, e.g. connection/timeout errors
-    or cases where recording status is not really complete
-    """
-    pass
-
-
-class ApiResponseParsingFailure(Exception):
-    pass
-
-
 @setup_logging
 def handler(event, context):
     """
     This function receives an event on each new entry in the download urls
     DyanmoDB table
     """
 
-    ignore_schedule = event.get('ignore_schedule', False)
-    override_series_id = event.get('override_series_id')
-
-    sqs = boto3.resource('sqs')
-    download_queue = sqs.get_queue_by_name(QueueName=DOWNLOAD_QUEUE_NAME)
+    ignore_schedule = event.get("ignore_schedule", False)
+    override_series_id = event.get("override_series_id")
+    sqs = boto3.resource("sqs")
 
-    # find a message from the downloads queue
-    try:
-        messages = download_queue.receive_messages(
-            MaxNumberOfMessages=1,
-            VisibilityTimeout=700
+    # try DOWNLOAD_MESSAGES_PER_INVOCATION number of times to retrieve
+    # a recoreding that matches the class schedule
+    download_message = None
+    for _ in range(int(DOWNLOAD_MESSAGES_PER_INVOCATION)):
+        download_message = Download(
+            sqs, ignore_schedule, override_series_id
         )
-        download_message = messages[0]
-        logger.info({'queue message': download_message})
-    except IndexError:
-        logger.info("No download queue messages available")
-        return
+        if not download_message.body:
+            logger.info("No messages available in downloads queue.")
+            return
+        if download_message.opencast_series_id:
+            break
```

Comment:
No, because in this case the message has matched with the schedule so you will continue to process the message. The message will be deleted after sending a message to the upload queue or deadletter queue.